### PR TITLE
Improve search

### LINKED ISSUE #302: Improve search UX
Try out the search and make proposals how to improve it.

Hints:

* search syntax is Lucene style, which is precise but uncommon if compared to Google search
* where should the results appear?
* Should the results disappear after click?

### PR DESCRIPTION
**Pull request by bonndan**
closes #302 

* various improvements of search term handling
* improved facets

### CODE REVIEW

File: src/main/app/src/Components/Landscape/Search/Search.tsx [hunk 1]
```diff
@@ -54,56 +64,122 @@ const Search: React.FC<PropsInterface> = ({ setSidebarContent, ...props }) => {
...
 
+    const help = (
+      <div>
+        <h3>Search</h3>
+        Start typing to match items having name parts or description starting with the given term.
+        Find everything starting with 'foo':
+        <br />
+        <pre>foo</pre>
+        <br />
+        Use whitespaces as AND condition (default). Search in name and description for both words:
```
Comment (mfbieber) [inline src/main/app/src/Components/Landscape/Search/Search.tsx:131]: What would be the description? 

I tried to select several Wordpress items by typing `wor c` in http://localhost:3000/#/landscape/nivio:example and I was expecting to select wordpress-web and wordpress-redis and wordpress-mariadb, but it only selects wordpress-redis. They belong to the groups"content" and "common".
`wor co` does not find anything, for example. So the first word is the name of the item and the second is?

**Review by mfbieber — APPROVED** [review]
This improvement is great. I also like that you can select e.g. component, layer, etc. to find things 👍 

Just one question about the tooltip for the search (see in the comments), maybe we could make that clearer to understand.

File: src/main/app/src/Components/Landscape/Search/Search.tsx (hunk 1 above)
Comment (bonndan) [reply src/main/app/src/Components/Landscape/Search/Search.tsx:131]: Good point. The "multi field" query regards identifier, name and description. If you want to search in groups, you have to enter "group:co*". In addition, you query translates to "wor* AND c* in the three fields.

That reminds me to add facets for item.type and item.group.

Comment (bonndan) [reply src/main/app/src/Components/Landscape/Search/Search.tsx:131]: I've added #498 as follow-up

Comment (bonndan) [reply src/main/app/src/Components/Landscape/Search/Search.tsx:131]: + #500 as follow-up

**Pull request merged**